xen/arch/arm/tee/ffa.c

```diff
@@ -221,7 +236,11 @@ static int ffa_domain_init(struct domain *d)
 {
     struct ffa_ctx *ctx;
 
-    if ( !ffa_version )
+     /*
+      * We can't use that last possible domain ID or get_vm_id() would cause
+      * an overflow.
+      */
+    if ( !ffa_version || d->domain_id == UINT16_MAX)
         return -ENODEV;
```

Comment:
This will prevent the guest from being created and we should say why.
I would suggest to return something different than ENODEV (EINVAL ?) and add an error print.
We might need to discuss this with the community, either domain_id 0 could be left unused by Xen or UINT16_MAX could be.